lib/debug/source_repository.rb

```diff
@@ -49,11 +49,13 @@ def add_path path
     def get_si iseq
       return unless iseq
 
+      path = iseq.absolute_path
```

Comment:
I prefer https://github.com/ruby/debug/pull/55/files#diff-f199a045ec16c0d96b45c5260edd5f1cc070f5e6d9d30a9ec6d989b9b4a59178L54 with `elsif path` instead of `else`.

Reply:
and is it independent patch?

Reply:
> with elsif path instead of else
sorry I don't get it. I'm using `elsif path`, do you want me to change back to `else`?

> and is it independent patch?
yes

Reply:
initially I submitted this PR for this patch, but then the CI failed because of the other 2 issues. so I had to fix them at once.

Reply:
I prefer:

```ruby
      elsif @files.has_key?(path = iseq.absolute_path)
        @files[path]
      elsif path
        add_path(path)
```

because it can reduce `iseq.absolute_path` call.

Reply:
I see, pushed 👍